### ISSUE TITLE
Allow deployments to ignore storage when loading flows

### ISSUE THREAD

**Issue by marwan116** [PrefectHQ/prefect#7600]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

As it currently stands, when loading a flow from a flow run, the following function [load_flow_from_flow_run](https://github.com/PrefectHQ/prefect/blob/33eb3f02fd1ff151c37155c6c04e811853580723/src/prefect/deployments.py#L141) is called.

```python
async def load_flow_from_flow_run(
    flow_run: schemas.core.FlowRun, client: OrionClient, ignore_storage: bool = False
) -> Flow:
    """
    Load a flow from the location/script provided in a deployment's storage document.
    If `ignore_storage=True` is provided, no pull from remote storage occurs.  This flag
    is largely for testing, and assumes the flow is already available locally.
    """
    deployment = await client.read_deployment(flow_run.deployment_id)

    if not ignore_storage:
        if deployment.storage_document_id:
            storage_document = await client.read_block_document(
                deployment.storage_document_id
            )
            storage_block = Block._from_block_document(storage_document)
        else:
            basepath = deployment.path or Path(deployment.manifest_path).parent
            storage_block = LocalFileSystem(basepath=basepath)

        sys.path.insert(0, ".")
        await storage_block.get_directory(from_path=deployment.path, local_path=".")

    flow_run_logger(flow_run).debug(
        f"Loading flow for deployment {deployment.name!r}..."
    )

    import_path = relative_path_to_current_platform(deployment.entrypoint)

    # for backwards compat
    if deployment.manifest_path:
        with open(deployment.manifest_path, "r") as f:
            import_path = json.load(f)["import_path"]
            import_path = (
                Path(deployment.manifest_path).parent / import_path
            ).absolute()
    flow = await run_sync_in_worker_thread(import_object, str(import_path))
    return flow
```

The function takes an `ignore_storage` boolean flag, however, as far as I am aware, there currently is no way to set this flag when building and applying a Deployment. 

Therefore what ends up happening is if no storage block is set, i.e. `deployment.storage_document_id` is None, a default LocalFileSystem storage block is created on the fly here `storage_block = LocalFileSystem(basepath=basepath)`

`storage_block.get_directory` is then called 

see this line
`storage_block.get_directory(from_path=deployment.path, local_path=".")`

In the case of a `LocalFileSystem` storage block, `get_directory` will attempt to copy the deployment path to the local path using [shutil.copytree](https://github.com/PrefectHQ/prefect/blob/45dd44b3ec27f9b2a13eb43a56406dd1042ff259/src/prefect/filesystems.py#L144)

And if the `src` path does not exist this will raise an error.

In the case no deployment path is set - then [it will default to `opt/prefect/flows`](https://github.com/PrefectHQ/prefect/blob/main/src/prefect/deployments.py#L721)

The final step is to import the flow 
``` 
   import_path = relative_path_to_current_platform(deployment.entrypoint)
   ...
   flow = await run_sync_in_worker_thread(import_object, str(import_path))
```

### Describe the proposed behavior

We are in the process of migrating some flows from prefect1 to prefect2. We intend to store our flows on the docker image that is being run by our `KuberentesJob` infrastructure block. Therefore in our case, the flow is directly available on the image - i.e. directly performing a python module import without fetching any files from storage should work. (in prefect 1 this was known as `ModuleStorage`)

What is inconvenient is that the default path `opt/prefect/flows` is not available on the docker container. So in order to get our deployments to work, our current workaround is to pass a dummy path when creating our Deployment which we know will exist on most of the docker containers that our team will create ...  

This is not perfect given it assumes all docker images will have the dummy path available. It also involves an unnecessary copy operation. We would like instead to be able to pass a flag to ignore using any storage block when loading our flow from the flow run.

### Example Use

A user would create a deployment like so

```python
from prefect.deployments import Deployment
...
deployment = Deployment(
    ...,
    ignore_storage=True
 )
```

the CLI command would probably be 
```
prefect deployment build {my_deployment} ... --ignore-storage --skip-upload
```

Currently, we have to do the following:
```python
from prefect.deployments import Deployment
....

deployment = Deployment(
    ...,
    path="/my/dummy/path",  # a necessary hack to get prefect to work
)
```

### Additional context

We understand the tradeoff of the inconvenience of having to mint an image every time a flow change is introduced versus the easiness of copying our code onto a remote storage like AWS S3. 

We however prefer this approach given our flows' images are thin layers on top of a relatively stable base image, and we would like to keep our flows as part of a python package that is in turn containerized.

**Comment by james-tisato** [comment]
+1 for this feature. We have been "deploying" a dummy directory to storage (and having the agent pull it down to run a flow) just to avoid errors, and then deploying our code separately in a way that better suits our processes (in our case, installing into an externally managed virtual environment). It would be great not to need this step.

**Comment by tjgalvin** [comment]
I also was bitten by this recently, and have resorted to the same set work around as @james-tisato .

**Comment by james-tisato** [comment]
We'd still like this feature.

**Comment by cicdw** [comment]
I believe this will be resolved by the new projects structure which is getting released in beta tomorrow with 2.9.1.  

In this setup, there's no need for storage blocks at all (which really get in the way for debugging / customizing).  Once the release is out you can experiment with this using `prefect project init --recipe docker` which I believe will capture the local-to-docker setup and allow you to customize it further.

To be super explicit, running that command will produce a `prefect.yaml` file that you'll want to edit:

```yaml
# File for configuring project / deployment build, push and pull steps

# Generic metadata about this project
name: hello-projects
prefect-version: 2.9.1

# build section allows you to manage and build docker images
build:
- prefect_docker.projects.steps.build_docker_image:
    requires: "prefect-docker>0.1.0"
    image_name: "your-image-name-here"
    tag: "your-tag-here"
    dockerfile: "path-to-dockerfile-here"

push: null

# runtime instructions to clone project
pull:
- prefect.projects.steps.set_working_directory:
    # EDIT THIS TO YOUR LOCATION WITHIN THE IMAGE
    directory: /opt/prefect/hello-projects
```

Until the docs are built and published tomorrow, you can reference the docs within Github [here](https://github.com/PrefectHQ/prefect/blob/main/docs/concepts/projects.md)
